docs/madr/decisions/056-meshexternalservice-and-zoneegress.md

```diff
@@ -0,0 +1,406 @@
+
+###### New behaviour
+
+If on sidecar:
+- limit is multiplied by the number of instances (which is bad)
+- less traffic in the mesh (traffic does )
+
+###### Verdict
+
+It seems that it only makes sense on egress, we don't want the ratelimit to be changing with number of instances.
+
+##### MeshRetry
+
+###### Old ExternalService behaviour
+
+Both Retry and MeshRetry are configured on the sidecar
+
+###### New behaviour
+
+If on egress:
+- when an ExternalService is unavailable a retry will only occur `numRetries` tries - not `numRetries` * instances (in sidecar case) - which is good because we're not overwhelming the service
+- same story as circuit breaker in terms of acting as an aggregate / sensitivity
+- we definitely shouldn't configure it both on sidecar and egress, that way we'll have squared retries
+
+##### MeshTimeout
+
+###### Old ExternalService behaviour
+
+Both Timeout and MeshTimeout are configured on the sidecar.
+
+###### New behaviour
+
+If on egress:
+- we would have to match the timeouts from the sidecar not to have a clash (that server times out before client)
+- I don't see any advantages of having this on egress but nothing rules it out
+
+##### MeshTrace
+
+###### Old ExternalService behaviour
+
+Both TrafficTrace and MeshTrace do not support egress.
+
+###### New behaviour
+
+I think it makes sense both on sidecar and egress.
+If we're troubleshooting something it would be great to see spans from sidecar going to ExternalService.
+
+##### MeshTrafficPermission
+
+###### Old ExternalService behaviour
+
+Both MeshTrafficPermission and TrafficPermission only works with egress.
+
+###### New behaviour
+
+It is already implemented on egress for MeshTrafficPermission.
+
+If on the sidecar:
+- could you circumvent this limitation by trying to directly call through egress?
+
+##### MeshLoadBalancingStrategy
+
+###### Old ExternalService behaviour
+
+There is no old behaviour because there was no concept of Endpoints in External Service
+
+###### New behaviour
+
+Only the `loadBalancer` field makes sense in this regard.
+
+If on sidecar:
+- if you don't need/want egress it still makes sense to pick load balancing mechanism
+
+If on egress:
+- makes the same sense on as on sidecar
+
+##### MeshTCPRoute and MeshHTTPRoute
+
+###### Old ExternalService behaviour
+
+It worked for both egress and sidecar.
+
+###### New behaviour
+
+I don't see why we would change that behaviour.
+
+#### For outbound policies on the sidecar, for inbound policies on egress
+
+It naturally fits our targetRef mechanism.
+It does not make sense to apply inbound policies on the sidecar since there is no inbound for it.
+Egress plays a role of a "sidecar" that is inside the mesh, and we can target it's inbound.
+Configuring outbound policies on the sidecar avoids the problem with different clients influencing each other (see [MeshCircuitBreaker](#meshcircuitbreaker)).
+
+##### How to configure it
+
+###### From only policy (policies that do not have to)
```

Comment:
```suggestion
###### `from` only policy (policies that do not have `to`)
```

I find it quite exhausting to read to and from everywhere if they aren't formatted, they're too common as normal words

actually

```suggestion
###### `from` only policy
```